modules/framework/galasa-parent/dev.galasa.framework.resource.management/src/main/java/dev/galasa/framework/resource/management/internal/MonitorConfiguration.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright contributors to the Galasa project
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package dev.galasa.framework.resource.management.internal;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.regex.Pattern;
+import java.util.regex.PatternSyntaxException;
+
+import dev.galasa.framework.spi.FrameworkException;
+
+public class MonitorConfiguration {
+    
+    private String stream;
+    private List<Pattern> includesRegexList;
+    private List<Pattern> excludesRegexList;
+
+    public MonitorConfiguration(String stream, List<String> includesRegexList, List<String> excludesRegexList) throws FrameworkException {
+        this.stream = stream;
+        this.includesRegexList = convertListToPatternList(includesRegexList);
+        this.excludesRegexList = convertListToPatternList(excludesRegexList);
+    }
+
+    public String getStream() {
+        return stream;
+    }
+
+    public List<Pattern> getIncludesRegexList() {
+        return includesRegexList;
+    }
+
+    public List<Pattern> getExcludesRegexList() {
+        return excludesRegexList;
+    }
+
+    private List<Pattern> convertListToPatternList(List<String> regexList) throws FrameworkException {
```

Comment:
Should this pattern compilation be moved somewhere else?

Reply:
Possibly, some class that is passed the includes/excludes, then you can ask it isThisClassAllowed(xxx) or something ? Then we can test it in isolation.

Reply:
Moved into a MonitorFilter class with unit tests to accompany it in https://github.com/galasa-dev/galasa/pull/236/commits/fa36d081355b9cd4e5ef366e28cde2f8089f0b53